File: forumClient/src/Dto/app/TopicInfo.ts
import type { UserBasicInfo } from "../UserInfo"
import type { ThreadBasicInfo } from "./ThreadInfo"

export type TopicEntry = {
    topicId: string,
    boardId: string,
    topicName: string,
    description: string,
    createdByUserId: string,
    createdDate: Date
}

export type TopicBasicInfo = {
    topic: TopicEntry,
    totalThreads: number,
    totalPosts: number,
    numTotalThreads: number,
    numNewThreads: number,
    newestThread: ThreadBasicInfo,
    createdByUser: UserBasicInfo
}

export type TopicFullInfo = {
    topic: TopicEntry,
    totalPosts: number,
    numTotalThreads: number,
    numNewThreads: number,
    newestThread: ThreadBasicInfo,
    threads: ThreadBasicInfo[],
    totalThreads: number,
    createdByUser: UserBasicInfo
}

export type CreateTopicRequest = {
    boardId: string,
    topicName: string,
    description: string,
}

export type TopicSummary = {
    totalTopics: number
}

File: server/RestApiServer/Dto/Forum/TopicInfo.cs
using RestApiServer.Db;
using RestApiServer.Dto.App;

namespace RestApiServer.Dto.Forum
{
    public class TopicBasicInfo
    {
        public required TopicEntry Topic { get; set; }
        public required int TotalThreads { get; set; } = 0;
        public required int TotalPosts { get; set; } = 0;
        public required int NumTotalThreads { get; set; }
        public int NumNewThreads { get; set; }
        public ThreadBasicInfo NewestThread { get; set; } = null!;
        public UserBasicInfo CreatedByUser { get; set; } = null!;
    }

    public class TopicFullInfo
    {
        public required TopicEntry Topic { get; set; }
        public required int TotalThreads { get; set; } = 0;
        public required int TotalPosts { get; set; } = 0;
        public required int NumTotalThreads { get; set; }
        public int NumNewThreads { get; set; }
        public ThreadBasicInfo NewestThread { get; set; } = null!;
        public required UserBasicInfo CreatedByUser { get; set; }
        public required List<ThreadBasicInfo> Threads { get; set; }
    }
    public class TopicSummary 
    {
        public required int TotalTopics { get; set; } = 0;
    }
}